tools/RazorPageGenerator/Program.cs

```diff
@@ -0,0 +1,137 @@
+            var fileName = Path.GetFileName(cshtmlFilePath);
+            var fileNameNoExtension = Path.GetFileNameWithoutExtension(fileName);
+            var codeLang = new CSharpRazorCodeLanguage();
+            var host = new RazorEngineHost(codeLang);
+            host.DefaultBaseClass = "Microsoft.Extensions.RazorViews.BaseView";
+            host.GeneratedClassContext = new GeneratedClassContext(
+                executeMethodName: GeneratedClassContext.DefaultExecuteMethodName,
+                writeMethodName: GeneratedClassContext.DefaultWriteMethodName,
+                writeLiteralMethodName: GeneratedClassContext.DefaultWriteLiteralMethodName,
+                writeToMethodName: "WriteTo",
+                writeLiteralToMethodName: "WriteLiteralTo",
+                templateTypeName: "HelperResult",
+                defineSectionMethodName: "DefineSection",
+                generatedTagHelperContext: new GeneratedTagHelperContext());
+            var engine = new RazorTemplateEngine(host);
+
+            using (var fileStream = File.OpenText(cshtmlFilePath))
+            {
+                var code = engine.GenerateCode(
+                    input: fileStream,
+                    className: fileNameNoExtension,
+                    rootNamespace: Path.GetFileName(rootNamespace),
+                    sourceFileName: fileName);
+
+                var source = code.GeneratedCode;
+                var startIndex = 0;
+                while (startIndex < source.Length)
+                {
+                    var startMatch = @"<%$ include: ";
+                    var endMatch = @" %>";
+                    startIndex = source.IndexOf(startMatch, startIndex);
+                    if (startIndex == -1)
+                    {
+                        break;
+                    }
+                    var endIndex = source.IndexOf(endMatch, startIndex);
+                    if (endIndex == -1)
+                    {
+                        break;
+                    }
+                    var includeFileName = source.Substring(startIndex + startMatch.Length, endIndex - (startIndex + startMatch.Length));
+                    includeFileName = SanitizeFileName(includeFileName);
+                    Console.WriteLine("      Inlining file {0}", includeFileName);
+                    var replacement = File.ReadAllText(Path.Combine(basePath, includeFileName)).Replace("\"", "\\\"").Replace("\n", "\\n").Replace("\r", "\\r");
+                    source = source.Substring(0, startIndex) + replacement + source.Substring(endIndex + endMatch.Length);
+                    startIndex = startIndex + replacement.Length;
+                }
+
+                // Make the generated class 'internal' instead of 'public'
+                source = source.Replace("public class", "internal class");
+
+                File.WriteAllText(Path.Combine(basePath, string.Format("{0}.cs", fileNameNoExtension)), source);
```

Comment:
Any good reason to change this? Just consistency?

---
In reply to: [75028383](https://github.com/aspnet/Common/pull/139#discussion_r75028383) [](ancestors = 75028383)